homeassistant/components/http/forwarded.py

```diff
@@ -0,0 +1,98 @@
+"""Middleware to handle forwarded data by a reverse proxy."""
+from ipaddress import ip_address
+import logging
+
+from aiohttp.hdrs import X_FORWARDED_FOR, X_FORWARDED_HOST, X_FORWARDED_PROTO
+from aiohttp.web import HTTPBadRequest, middleware
+
+from homeassistant.core import callback
+
+_LOGGER = logging.getLogger(__name__)
+
+# mypy: allow-untyped-defs
+
+
+@callback
+def setup_forwarded(app, use_x_forwarded_for, trusted_proxies):
+    """Create IP Ban middleware for the app."""
+
+    @middleware
+    async def forwarded_middleware(request, handler):
+        overrides = {}
+
+        # Handle X-Forwarded-For
+        forwarded_for = request.headers.getall(X_FORWARDED_FOR, [])
+        if not forwarded_for:
+            # No forwarding headers, continue as normal
+            return await handler(request)
+
+        # Multiple X-Forwarded-For headers
+        if len(forwarded_for) > 1:
+            _LOGGER.error("Too many headers for X-Forwarded-For", extra=request.headers)
+            raise HTTPBadRequest
+
+        # Process IP addresses in the forwarded for header
+        forwarded_for = list(reversed(forwarded_for[0].split(",")))
+        try:
+            forwarded_for = [
+                ip_address(addr) for addr in (a.strip() for a in forwarded_for) if addr
```

Comment:
```suggestion
                ip_address(addr.strip()) for addr in forwarded_for if addr
```

Why the `if addr` ? This allows us to validate `1.1.1.1, , 1.2.3.4` as a valid value but it looks incorrect?

Reply:
Good point, it should fail in that case. Good test case to add to the tests as well.

Reply:
Test case added in 6137099